kasmvnc/run.sh

```diff
@@ -14,166 +15,169 @@ check_installed() {
-          exit 1
-          ;;
-      esac
+      bin_name="kasmvncserver_$${codename}_${KASM_VERSION}_$${arch}.deb"
+      install_deb "$base_url/$bin_name"
       ;;
-    oracle)
-      if [[ "$version" == "8" ]]; then
-        install_rpm_oracle8 "https://github.com/kasmtech/KasmVNC/releases/download/v${VERSION}/kasmvncserver_oracle_8_${VERSION}_$${arch}.rpm"
-      else
-        echo "Unsupported Oracle version: $${version}"
-        exit 1
-      fi
-      ;;
-    centos)
-      if [[ "$version" == "7" ]]; then
-        install_rpm_centos7 "https://github.com/kasmtech/KasmVNC/releases/download/v${VERSION}/kasmvncserver_centos_core_${VERSION}_$${arch}.rpm"
-      else
-        install_rpm "https://github.com/kasmtech/KasmVNC/releases/download/v${VERSION}/kasmvncserver_centos_core_${VERSION}_$${arch}.rpm"
-      fi
+    oracle | fedora | opensuse)
+      bin_name="kasmvncserver_$${distro}_$${distro_version}_${KASM_VERSION}_$${arch}.rpm"
+      install_rpm "$base_url/$bin_name"
       ;;
     alpine)
-      if [[ "$version" == "3.17" || "$version" == "3.18" || "$version" == "3.19" || "$version" == "3.20" ]]; then
-        install_alpine "https://github.com/kasmtech/KasmVNC/releases/download/v${VERSION}/kasmvnc.alpine_$${version}_$${arch}.tgz"
-      else
-        echo "Unsupported Alpine version: $${version}"
-        exit 1
-      fi
-      ;;
-    fedora | opensuse)
-      install_rpm "https://github.com/kasmtech/KasmVNC/releases/download/v${VERSION}/kasmvncserver_$${distro}_$${version}_${VERSION}_$${arch}.rpm"
+      bin_name="kasmvnc.alpine_$${distro_version//./}_$${arch}.tgz"
+      install_alpine "$base_url/$bin_name"
       ;;
     *)
-      echo "Unsupported distribution: $${distro}"
+      echo "Unsupported distribution: $distro"
       exit 1
       ;;
   esac
 else
   echo "vncserver already installed. Skipping installation."
 fi
 
-# Coder port-forwarding from dashboard only supports HTTP
-sudo bash -c "cat > /etc/kasmvnc/kasmvnc.yaml <<EOF
+tee "$HOME/.vnc/kasmvnc.yaml" > /dev/null << EOF
 network:
   protocol: http
   websocket_port: ${PORT}
   ssl:
     require_ssl: false
+    pem_certificate:
+    pem_key:
   udp:
     public_ip: 127.0.0.1
-EOF"
+EOF
 
 # This password is not used since we start the server without auth.
 # The server is protected via the Coder session token / tunnel
 # and does not listen publicly
-echo -e "password\npassword\n" | vncpasswd -wo -u $USER
+echo -e "password\npassword\n" | vncpasswd -wo -u "$USER"
 
 # Start the server
 printf "🚀 Starting KasmVNC server...\n"
-sudo -u $USER bash -c "vncserver -select-de ${DESKTOP_ENVIRONMENT} -disableBasicAuth" > /tmp/kasmvncserver.log 2>&1 &
+vncserver -select-de "${DESKTOP_ENVIRONMENT}" -disableBasicAuth > /tmp/kasmvncserver.log 2>&1 &
+
+# Wait for server to start
+sleep 5
+if ! pgrep -f vncserver > /dev/null; then
+  echo "ERROR: Failed to start KasmVNC server. Check logs at /tmp/kasmvncserver.log"
```

Comment:
Should we print last few lines from log for ease of debugging?

Reply:
I just checked, the log file is fairly short, we can probably print the whole thing.

```
$ cat /tmp/kasmvncserver.log 

New 'qlcplus-v4-workspace:1 (coder)' desktop is qlcplus-v4-workspace:1

Users configured:
coder (can use keyboard and mouse, add/remove users)

Log file is /home/coder/.vnc/qlcplus-v4-workspace:1.log

Starting applications specified in /home/coder/.vnc/xstartup

Paste this url in your browser:
https://10.88.0.67:6800
```

Reply:
I added this to the exit block, wondering if we want to just print this by default after the sleep?
There is some good information in here and it is short.

Reply:
I worry it could be _a lot_ longer if there were a slew of errors, but it's hard to predict. I'd probably do:
```
echo "Showing the last 5 log lines, inspect the full log at /tmp/kasmvncserver.log:"
tail -n5 /tmp/kasmvncserver.log
```
But I wouldn't reject the PR if you decide it's better to print it all. 😄

Reply:
> I added this to the exit block, wondering if we want to just print this by default after the sleep? There is some good information in here and it is short.

That's an interesting idea, feel free to play around with it.

Reply:
It looks good on a successful run, not sure what an error will show, but I limited it to 10 non-blank lines.

```
🚀 Starting KasmVNC server...
New 'qlcplus-v4-workspace:1 (coder)' desktop is qlcplus-v4-workspace:1
Users configured:
coder (can use keyboard and mouse, add/remove users)
Log file is /home/coder/.vnc/qlcplus-v4-workspace:1.log
Starting applications specified in /home/coder/.vnc/xstartup
Paste this url in your browser:
https://10.88.0.72:6800
🚀 KasmVNC server started successfully!
```